cgen error, when doing `x := opt()[0] or { 1 }`, and `fn opt() ?[]int`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.3 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/cleanv
vmodules: /home/delian/.vmodules
vroot: /v/cleanv
vexe: /v/cleanv/v
vexe mtime: 2022-03-20 17:45:59
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 4d99157.71edaa0

Git version: git version 2.25.1
Git vroot status: weekly.2022.11-38-g71edaa07
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg a.v`
```v
fn abc() ?[]int {
	return [1, 2, 3]
}

fn main() {
	a := abc()[0] or { 5 }
	dump(a)
}

```

**What did you expect to see?**

a checker error, that I have not handled the optional return value of abc(), before the indexing [0]


**What did you see instead?**
```
==================
/tmp/v_1000/a.12860926113972852713.tmp.c:11418: error: cannot convert 'struct Option_Array_int' to 'struct array'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -3775,6 +3775,9 @@ pub fn (mut c Checker) index_expr(mut node ast.IndexExpr) ast.Type {
 		&& typ !in [ast.byteptr_type, ast.charptr_type] && !typ.has_flag(.variadic) {
 		c.error('type `$typ_sym.name` does not support indexing', node.pos)
 	}
+	if typ.has_flag(.optional) {
+		c.error('type `?$typ_sym.name` is optional, it does not support indexing', node.left.pos())
+	}
 	if typ_sym.kind == .string && !typ.is_ptr() && node.is_setter {
 		c.error('cannot assign to s[i] since V strings are immutable\n' +
 			'(note, that variables may be mutable but string values are always immutable, like in Go and Java)',

--- a/vlib/v/checker/tests/index_of_optional_err.out
+++ b/vlib/v/checker/tests/index_of_optional_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/index_of_optional_err.vv:6:7: error: type `?[]int` is optional, it does not support indexing
+    4 |
+    5 | fn main() {
+    6 |     a := abc()[0] or { 5 }
+      |          ~~~~~
+    7 |     dump(a)
+    8 | }

--- a/vlib/v/checker/tests/index_of_optional_err.vv
+++ b/vlib/v/checker/tests/index_of_optional_err.vv
@@ -1,0 +1,8 @@
+fn abc() ?[]int {
+	return [1, 2, 3]
+}
+
+fn main() {
+	a := abc()[0] or { 5 }
+	dump(a)
+}

